src/main.rs

```diff
@@ -125,13 +177,101 @@ fn main() {
+                let ref controllers = program.state.lock().unwrap().controllers;
+                for controller in controllers {
+                    print_controller(&controller);
+                }
+            }
+            "controllers_dbg" => {
+                let ref controllers = program.state.lock().unwrap().controllers;
+                for controller in controllers {
+                    println!("{:?}\n", controller);
+                }
+            }
+            "nodes" => {
+                let mut program_state = program.state.lock().unwrap();
+                let ref mut nodes_map = program_state.nodes_map;
+                for (ref controller, ref mut node_set) in nodes_map {
+                    print_controller(&controller);
+                    for node in node_set.iter() {
+                        println!("  Node: {:3} {:17} {:30} {:50} {:30} {:30}",
+                                 node.get_id(),
+                                 node.get_basic_as_str(),
+                                 node.get_type(),
+                                 node.get_product_name(),
+                                 node.get_name(),
+                                 node.get_location()
+                                 );
+                    }
+                }
+            }
+            "nodes_dbg" => {
+                let ref nodes = program.state.lock().unwrap().nodes;
+                for node in nodes {
+                    println!("{:?}\n", node);
+                }
+            }
+            "values" => {
+                let ref value_ids = program.state.lock().unwrap().value_ids;
+                for value_id in value_ids {
+                    if value_id.get_genre() != ValueGenre::ValueGenre_User {
+                        continue;
+                    }
+                    let node = Node::from_id(value_id.get_home_id(), value_id.get_node_id());
+                    let val_str = match value_id.as_string() {
+                        Ok(str) => str,
+                        Err(str) => String::from(str)
+                    };
+                    println!(" HomeId: {:08x} ID: {:016x} NodeId: {:3} {:20} CC: {:3} Type: {:8} Label: {:20} Value: {}",
+                             value_id.get_home_id(),
+                             value_id.get_id(),
+                             value_id.get_node_id(),
+                             node.get_name(),
+                             value_id.get_command_class_id(),
+                             value_id.get_type_as_str(),
+                             value_id.get_label(),
+                             val_str);
+                }
+            }
+            "values_dbg" => {
+                let ref value_ids = program.state.lock().unwrap().value_ids;
+                for value_id in value_ids {
+                    println!("{:?}", value_id);
+                }
+            }
+            _ => {
+                println!("Unrecognized command: '{}'", tokens[0]);
```

Comment:
Since this was just an example program I wasn't too concerned about stuff like help.

Once we have a proper adapter, then we can write a python front-end and it will get free help for all of the commands.